### PR TITLE
Adding enhancements for Zones in moc

### PR DESCRIPTION
**Pull request by mkatturu**

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -1,945 +1,21 @@
-cloud.google.com/go v0.34.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (msftpgayam) [inline go.sum:1]: Any idea why a small change to the health state enum caused so many changes here?

File: rpc/common/moc_common_common.proto [hunk 2]
```diff
@@ -56,6 +56,7 @@ enum HealthState {
 	DEGRADED = 5;
 	// The entity went missing from the agent
 	NOTFOUND = 6;
+	NOTREADY = 7;
```
Comment (msftpgayam) [inline rpc/common/moc_common_common.proto:59]: Is there a doc to describe what each health state means for zones?

Comment (mkatturu) [reply rpc/common/moc_common_common.proto:59]: Yes this is defined in our zones feature design doc. 

https://github.com/microsoft/wssdagent/blob/master/doc/features/zones/Zone_Management_in_moc.md

image

**Review by piyushka17 — DISMISSED** [review]

File: go.sum (hunk 1 above)
Comment (mkatturu) [reply go.sum]: I updated my branch and go sum is now updated and almost all the changes in go sum are reverted

**Comment by jocardeMSFT** [comment]
General Comment, request you add more to the PR description including change summary, change purpose, and testing complete. From a maintainability point of view, it helps future devs understand the intent of your changes and adds context for why things were implemented a certain way.

**Review by jocardeMSFT — APPROVED** [review]

File: go.mod [hunk 3]
```diff
@@ -20,7 +20,7 @@ require (
 	golang.org/x/net v0.29.0 // indirect
```
Comment (piyushka17) [inline go.mod:20]: I think you shouldn't have any change in go.mod, go.sum. You can possibly revert these files?

**Review by piyushka17 — APPROVED** [review]

**Pull request merged**